### ISSUE TITLE
Forced OCR on CBZ files not working

### ISSUE THREAD

**Issue by Gabriel85** [koreader/koreader#11442]
* KOReader version: 2024.01
* Device: all

Now if you set "forced OCR" to ON with an CBZ file it does not seem to do anything. It would be a useful feature if  it works like with PDF files containing images. A CBZ is practically nothing else than a simplified image-only PDF. Therefore it should be handled the same as with PDF files. The detection probably would not work so well for detecting text in comic bubbles but certainly be very useful for printed text pages and mixed content. The same reasoning would apply for text reflow, but that you can't even enable at the moment.

If that is not possible for technical reasons, I think at least this option should be grayed out to avoid confusion.

**Comment by Gabriel85** [comment]
> It works for me. But note that probably by default panel zoom is enabled, which will take precedence over text selection. <img alt="" width="45%" src="https://private-user-images.githubusercontent.com/202757/302382025-49bdd9e8-7dfc-4f2c-90a1-3a4f7a40f3f7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDcxNTU4ODEsIm5iZiI6MTcwNzE1NTU4MSwicGF0aCI6Ii8yMDI3NTcvMzAyMzgyMDI1LTQ5YmRkOWU4LTdkZmMtNGYyYy05MGExLTNhNGY3YTQwZjNmNy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMjA1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDIwNVQxNzUzMDFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT00OTgwMjk2NTYyMTcwYjgzYWIxNGZiZTUzYmY1OGYzMjZhMzU4NWVkMmUwNmNiODgxMDhiMGYwMDg3ZDUwYmI1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.lhj9C3BQXpxtFJosO0r0n5D23c5bZes-d4o9wSvyWdY"> <img alt="" width="45%" src="https://private-user-images.githubusercontent.com/202757/302382178-b3436f9f-f930-4659-bbb6-b0855741ec7f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDcxNTU4ODEsIm5iZiI6MTcwNzE1NTU4MSwicGF0aCI6Ii8yMDI3NTcvMzAyMzgyMTc4LWIzNDM2ZjlmLWY5MzAtNDY1OS1iYmI2LWIwODU1NzQxZWM3Zi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMjA1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDIwNVQxNzUzMDFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT01OTNjZjhjYWY3MzdlOTE0ZTA2ODE0N2MxMmU5OTkxNDI1ZjUzYjVjZTMxZmI2ZjUyYTUzYTcwYzAwMjA1NjliJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.G0NaVrDn6Zfa4EnpbMxJJRoOvGKab1W1muZxXxX-104">

Yes thanks, it works perfectly well when panel zoom is disabled! Maybe "forced OCR" can automatically disable "panel zoom" or there could be an option to trigger the "panel zoom" with a double tap? Actually a double tap option would be really neat, because I trigger the "panel zoom" often by accident..

**Comment by Gabriel85** [comment]
There is another issue: "force OCR" is always enabled. Maybe it also affects PDFs. Would it be possible to just not do any OCR or text selection (if it is disabled) on image only documents?

**Comment by Frenzie** [comment]
If you keep them in separate folders you can create different folder defaults.

**Comment by Gabriel85** [comment]
> If you keep them in separate folders you can create different folder defaults.

"force OCR" is probably not something which anybody wants to have enabled with an CBZ by default. Its very useful on certain pages (with printed text) but useless for reading an actual comic. I think it would be best if it is only enabled when the switch is on (I guess that is why there is a switch) for CBZs.

But in at the very least I think there should be some indication that it is automatically disabled when "panel zoom" is enabled, to avoid confusion! Like this message you get when no OCR files are installed.

**Comment by Frenzie** [comment]
It's not enabled by default. If you have a document with a text layer at best you'd be wasting time by always forcing it.

**Comment by Gabriel85** [comment]
> It's not enabled by default. If you have a document with a text layer at best you'd be wasting time by always forcing it.

For me it is always enabled with CBZs, and it is impossible to have one with a text layer.

**Comment by Frenzie** [comment]
You can change the default back by long pressing on off.

**Comment by Gabriel85** [comment]
> You can change the default back by long pressing on off.

Yes I already have that as the default! Can it be that OCR is always enabled for non-text layer files?

**Comment by Frenzie** [comment]
A new default will only affect documents that were never opened before.

**Comment by Gabriel85** [comment]
> A new default will only affect documents that were never opened before.

I never had this set as default, and I just have tried it again with a file I have never opened and OCR is enabled, despite "force OCR" being disabled! Can somebody confirm that with an CBZ, please?

**Comment by Frenzie** [comment]
To be clear, forced OCR means to ignore the text layer. When there's no text layer the program automatically performs OCR on the area where you're long pressing. It's not doing anything if you're not.

**Comment by Gabriel85** [comment]
> To be clear, forced OCR means to ignore the text layer. When there's no text layer the program automatically performs OCR on the area where you're long pressing. It's not doing anything if you're not.

Yes that is what I am meaning! And it also doesn't work if "panel zoom" is activated and there is no indication of that in the app. So the setting is not only confusing, it is totally nonfunctioning when a CBZ Is opened. Maybe it would be better, if it overrides "panel zoom" or it it is just grayed out in that case?

And what about a double-click "panel zoom", so both can be enabled at the same time?

**Comment by Frenzie** [comment]
Yes, it being grayed out might be better, though I think you took what I wrote as exactly the opposite of what I wrote.

**Comment by Gabriel85** [comment]
> Yes, it being grayed out might be better, though I think you may have interpreted what I wrote as the reverse of what I wrote. OCR is always performed when there is no text layer, but only when you long-press.

Yes I get that! I just meant it would be nice if the user can disable it, since it doesn't work at all for comic pages and there is already a (non-working) setting for it. OCR detects gibberish, if you press long anywhere, even if there is nothing nearby.

I think the most ideal behaviour would be like that: OCR is disabled, unless "force OCR" is enabled. And if you enable "force OCR", OCR is enabled and panel zoom is automatically disabled. This would improve usability, without any change in the UI. There is already an quick setting for OCR, and "panel zoom" needs to be activated in the menu and I think both features are not needed at the same time anyways.
